resolver.go

```diff
@@ -332,6 +342,22 @@ func (r *Resolver) exchangeIP(ctx context.Context, host, ip, qname, qtype string
 		rmsg, dur, err = client.ExchangeWithConnContext(ctx, &qmsg, dconn)
 		conn.Close()
 	}
+	if r.tcpRetry && rmsg != nil && rmsg.MsgHdr.Truncated {
+		// Since we are doing another query, we need to recheck the deadline
+		if dl, ok := ctx.Deadline(); ok {
+			if start.After(dl.Add(-TypicalResponseTime)) { // bail if we can't finish in time (start is too close to deadline)
+				return nil, ErrTimeout
+			}
+			timeout = dl.Sub(start)
```

Comment:
This timeout value only ends up in the `log*` parameters below, where above it is used to configure the timeout of `client`. It appears that this should instead be 
```suggestion
			client.Timeout = dl.Sub(start)
```

Reply:
Yeah, good catch.
I also changed the log statements so it isn't incorrect after a TCP retry!